contest_manager.py

```diff
@@ -41,8 +23,81 @@ def __init__(self, client, dmoj_client : DmojClient):
         self._users_db = MicroDB("users")
         self._contests_db = MicroDB("contests")
 
+    '''
+    @commands.command()
+    async def force_add(self, ctx):
+        self._users_db.set("Sardor", "donkogronko")
+
     @commands.command()
-    async def create_contest(self, ctx, contest_name, active, registrationRequired, endDate, force, *problem_links):
+    async def print_db(self, ctx):
+        await ctx.send(json.dumps(self._contests_db.get_all(), indent=4))
+    '''
+    
+    @commands.command()
+    async def create_weekly(self, ctx, week_number, *problem_links):
+        contest_name = f"Weekly Problem Set #{week_number}"
+        end_date = datetime.datetime.now() + datetime.timedelta(days=7)
+        await self._create_contest(ctx, contest_name, True, False, end_date, False, *problem_links)
+
+    @commands.command()
+    async def end_weekly(self, ctx, week_number):
+        await self.end_contest(ctx, f"Weekly Problem Set #{week_number}")
+
+    @commands.command()
+    async def end_contest(self, ctx, contest_name):
+        if not contest_name in self._contests_db.get_all():
+            await ctx.send("There is no contest with this name")
+            return
+        
+        contest_dict = self._contests_db.get(contest_name)
+        users_dict =  self._users_db.get_all()
+        endDate = contest_dict["endDate"]
+        
+        def check(message):
+            return message.author.id == ctx.author.id and \
+                message.channel == ctx.channel
+
+        if contest_dict["active"] == False:
+            await ctx.send("This contest is already over.")
+            return
+
+        if datetime.datetime.now() < endDate:
+            date_string = endDate.strftime("%B %d, %Y")
+            await ctx.send(f"This contest is set to end on {date_string}. Are you sure you want to end the contest anyway?")
+            message = await self.client.wait_for("message", check=check)
+            if not message.content.lower() in ["y", "yes"]:
+                await ctx.send("Okay, aborted.")
+                return
+        
+        contest_dict["active"] = False
+        contest_dict["completion"], contest_dict["ranking"] = await self._contest_rankings(contest_name)
+        self._contests_db.set(contest_name, contest_dict)
+
+        await ctx.send(json.dumps(contest_dict["completion"], indent=4))
+        await ctx.send(contest_dict["ranking"])
+
+    async def _contest_rankings(self, contest_name):
```

Comment:
what does this return, method is not self explanatory from the name or the returned variable names.